src/components/ConversationSettings/LinkShareSettings.vue

```diff
@@ -21,36 +21,58 @@
 				{{ t('spreed', 'Allow guests to join this conversation via link') }}
 			</NcCheckboxRadioSwitch>
 
-			<NcCheckboxRadioSwitch v-show="isSharedPublicly"
-				:checked="isPasswordProtectionChecked"
-				:disabled="isSaving"
-				type="switch"
-				aria-describedby="link_share_settings_password_hint"
-				@update:checked="togglePassword">
-				{{ t('spreed', 'Password protection') }}
-			</NcCheckboxRadioSwitch>
-
-			<form v-if="showPasswordField" class="password-form" @submit.prevent="handleSetNewPassword">
-				<NcPasswordField ref="passwordField"
-					:value.sync="password"
-					autocomplete="new-password"
-					check-password-strength
+			<template v-if="isSharedPublicly">
+				<NcCheckboxRadioSwitch v-if="!forcePasswordProtection"
+					:checked="isPasswordProtectionChecked"
 					:disabled="isSaving"
-					class="password-form__input-field"
-					label-visible
-					:label="t('spreed', 'Enter new password')"
-					@valid="isValid = true"
-					@invalid="isValid = false" />
-				<NcButton :disabled="isSaving || !isValid"
-					type="primary"
-					native-type="submit"
-					class="password-form__button">
-					<template #icon>
-						<ArrowRight />
-					</template>
-					{{ t('spreed', 'Save password') }}
-				</NcButton>
-			</form>
+					type="switch"
+					aria-describedby="link_share_settings_password_hint"
+					@update:checked="togglePassword">
+					{{ t('spreed', 'Password protection') }}
+				</NcCheckboxRadioSwitch>
+				<p v-if="forcePasswordProtection && isPasswordProtectionChecked" class="app-settings-section__hint">
+					{{ t('spreed', 'This conversation is password-protected. Guests need password to join') }}
+				</p>
+				<NcNoteCard v-if="forcePasswordProtection && !isPasswordProtectionChecked && !isSaving"
+					type="warning">
+					{{ t('spreed', 'Password protection is needed for public conversations') }}
+					<NcButton class="warning__button" @click="enforcePassword">
```

Comment:
`type="primary"` to make more prominent? I barely notice it

![image](https://github.com/user-attachments/assets/3add28e3-807a-46af-9b03-4a3bfabb5abe)